osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -0,0 +1,164 @@
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getDlmsMasterKey(String deviceIdentification) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getDlmsAuthenticationKey(String deviceIdentification) {
+
+        try {
+            GetSecretsRequest request = getSoapRequestForKey(deviceIdentification, SecretType.E_METER_AUTHENTICATION_KEY);
+            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
+            Optional<TypedSecret> optionalTypedSecret = getTypedSecretFromSoapResponse(response, SecretType.E_METER_AUTHENTICATION_KEY);
+
+            return decryptOptionalSoapSecret(deviceIdentification, optionalTypedSecret);
+        }
+        catch(Exception e) {
+            LOGGER.error("Error while retrieving authentication key", e);
+        }
+        return new byte[0];
+    }
+
+    @Override
+    public byte[] getDlmsGlobalUnicastEncryptionKey(String deviceIdentification) {
+
+        try {
+            GetSecretsRequest request = getSoapRequestForKey(deviceIdentification, SecretType.E_METER_ENCRYPTION_KEY_UNICAST);
+            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
+            Optional<TypedSecret> optionalTypedSecret = getTypedSecretFromSoapResponse(response, SecretType.E_METER_ENCRYPTION_KEY_UNICAST);
+
+            return decryptOptionalSoapSecret(deviceIdentification, optionalTypedSecret);
+        }
+        catch(Exception e) {
+            LOGGER.error("Error while retrieving encryption key", e);
+        }
+        return new byte[0];
+    }
+
+    @Override
+    public byte[] getMbusDefaultKey(String mbusDeviceIdentification) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getMbusUserKey(String mbusDeviceIdentification) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] getDlmsPassword(String deviceIdentification) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public DlmsDevice storeNewKey(DlmsDevice device, byte[] encryptedKey, SecurityKeyType keyType) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public DlmsDevice validateNewKey(DlmsDevice device, SecurityKeyType keyType) throws ProtocolAdapterException {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] generateKey() {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] generateAndEncryptKey() {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public byte[] encryptMbusUserKey(byte[] mbusDefaultKey, byte[] mbusUserKey) throws ProtocolAdapterException {
+        throw new NotImplementedException();
+    }
+
+    private Optional<TypedSecret> getTypedSecretFromSoapResponse(GetSecretsResponse response, SecretType secretType) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "secretType". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1172)